### ISSUE TITLE
Can’t use in app purchase on tvOS 17

### ISSUE THREAD

**Issue by asuatton** [zipleen/tvheadend-iphone-client#289]
Impossible to restore the payed app.  When we go to « buy app » the app crash.   Please correct that quickly (as we payed for this app) 
Thanks

**Comment by zipleen** [comment]
hi
that code should work just fine. usually it's about environment.
try the following:
delete and install the app again
logout and login to icloud
make sure you don't have any security restrictions on your icloud settings
usually the first 2 fix the issue. i'm sorry but this is not really something I can fix because I can't replicate, but if it persists do let me know exact tvos version so I can try to reproduce it.

**Comment by asuatton** [comment]
I tryed everything. Log out login to iCloud.  No restriction. Re- install the app.  Reset my Apple TV and install TvhClient in first.  Nothing. Still can’t access to the purchase page.   Can you fix this by an update? Or push to my device the unlock ?   Why did you don’t make a free app and another pay app? I think it can be more reliable 
Thanks

**Comment by zipleen** [comment]
I don't have any ability to push any unlock to your device. I don't know what's wrong because I can't replicate your scenario.

are you able to tell me exactly the version of tvOS that you're experiencing these issues? 

you can try signing up for the beta version, which is unlocked by default https://testflight.apple.com/join/kNRWzusF 

2 versions don't make sense because I need people to attempt to connect to tvheadend first, in order to understand what tvheadend really is, and what this application is not. having 2 versions would allow people to buy and create confusion.

up until now, everyone that tried to buy the IAP, worked ok. the only thing I can do is attempt to debug this with your account, but that would require your credentials and I don't want to do that.

**Comment by asuatton** [comment]
Yes it’s true for the two versions ……. When I try your link for TestFlight, it say that the program doesn’t accept new members …. 
im on tvOS 17.1 build 21K69         On Apple TV 4K gen2 (2021) A2169
It was working like a charme on this Apple TV and this tvOS but it was already activated from preview tvOS 16. I had to reset my Apple TV and since tvos17 I can’t activate the app from restore purchase…. 
thanks

**Comment by zipleen** [comment]
a new beta version is up so you should be able to get this up and running. try signing up again.

but we still need to test this IAP flow in tvOS 17. I'll install tvOS 17 in one of my apple tvs so I can try the IAP flow in the current tvOS build

**Comment by asuatton** [comment]
Thanks a lot.  I try to sign up for beta just now and it still say that the program not accept new beta tester for the moment.

**Comment by mundschenk-at** [comment]
I've also seen "restore purchases" crashing the app, but it is "unlocked" afterwards (but also does not play TV anymore, the other features seem to work).

**Comment by zipleen** [comment]
looks like the issue has nothing to do with in app purchases, but rather the use of a non-admin user.
always debug issues by using an admin user and making sure the "admin user" option is selected

I'm closing this ticket because IAP works fine. it's usually something transient with the user device/account
